pkg/metrics/metrics.go

```diff
@@ -41,6 +43,20 @@ func CreateMetrics(reg prometheus.Registerer) Metrics {
 				Help:      "EKS Configmap watch failures",
 			},
 		),
+		StsConnectionFailure: factory.NewCounter(
+			prometheus.CounterOpts{
+				Namespace: Namespace,
+				Name:      "sts_connection_failures",
+				Help:      "Sts call could not succeed or timedout",
+			},
+		),
+		StsResponseErrorCodeNot200: factory.NewCounter(
```

Comment:
I think we should have a metric called `sts_responses_total` and use labels for the exact response code.  This is in line with the recommendations here https://prometheus.io/docs/practices/instrumentation/#use-labels. This will also be much more useful in the long run then only knowing if the code is not 200.

Reply:
I was following the metric convention from https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats due to my familiarity. We can use labels here